cookbooks/kinematically_driven_subduction_2d/doc/kinematically_driven_subduction_2d.md

```diff
@@ -100,3 +99,98 @@ boundary. This can also be seen in {numref}`fig:QQ_case1_results`.
 ```
+## Case 2a: Model with thermal evolution
+
+Case 2a builds on Case 1 by including a heterogeneous initial temperature field
+that combines the plate cooling model for the temperature in the plates with an
+adiabatic temperature gradient below the plates. For the plate cooling model,
+we assume that the overriding plate is 40 My old and the subducting plate 70 My.
+The temperature function is too complex to comfortably write out as a function
+expression in the input file, so it is put in a new initial temperature plugin
+called ’subduction plate cooling’. This plugin needs to be compiled as a shared
+library that can be called from the input file:
+
+```{literalinclude} Case2a_temperatures.prm
+
+```
+The initial temperature distribution can be seen in  {numref}`fig:QQ_case2a_temperature_t0`
+
+```{figure-md} fig:QQ_case2a_temperature_t0
+<img src="Case2a_temp_t0.*" />
+
+ Case 2a temperature distribution at time 0
+```
+
+The thermal expansivity is kept zero in Case 2a. Therefore the thermal evolution is
+not coupled to the mechanical evolution. A high conductivity (k = 183.33 Wm<sup>-1</sup>K<sup>-1</sup>)
+is defined for the upper mantle to enforce the mantle adiabat in the slow convection
+system.
+
+```{literalinclude} Case2a_materialmodel.prm
+
+```
+
+We monitor the temperature evolution by tracking the
+root-mean-square (RMS) temperature and the isotherm depth using
+two plugins called "composition_trms_statistics" and "isotherm depth".
+To this end, we add the postprocessors "composition RMS temperature statistics"
+and "isotherm depth" to the prm file. The RMS temperature is tracked
+over the subducting plate (comprising BOC_SP, SHB_SP, thermal_SP)
+and the whole domain over time. And the isotherm temperature to be
+tracked is prescribed as 800 &degC.
+
+```{literalinclude} Case2a_postprocessing.prm
+
+```
+
+{numref}`fig:QQ_case2a_diagnostics` shows the RMS temperature,
+800 &degC isotherm depth as well as the other diagnostics. The 800 &degC isotherm
+depth continuously increases over time, but towards the end at a slower rate.
+The RMS temperature of the whole domain keeps decreasing due to the
+input of cold subducting plate into the model domain. The slab RMS temperature
+decreases at the initial stage, then it goes up due to the slab being warmed up by the
+mantle. Note that the other diagnostics are indeed the same as for Case 1.
+
+```{figure-md} fig:QQ_case2a_diagnostics
+<img src="Case2a_diagnostics.*" />
+
+ ASPECT Case 2a and Case 1, Elefant Case 2a diagnostic quantities
+```
+
+## Case 2b: Model with a temperature-dependent density
+
+Case 2b builds on Case 2a.  Compared to Case 2a, the thermal expansivity in Case 2b
+is set to 2.5e-5 K<sup>-1</sup> instead of 0 K<sup>-1</sup>. Therefore, temperature feeds into the density
+and therefore into the Stokes equations. In addition, the benchmark uses different reference temperatures for each composition, but only one value can be specified for the reference temperature
```

Comment:
This renders the density (and therefore the Stokes equations) temperature-dependent.